modules/core/src/smithy4s/Transformation.scala

```diff
@@ -119,4 +119,30 @@ object Transformation {
       }
     }
 
+  implicit def mappedHintsServiceTransform[Alg[_[_, _, _, _, _]]]: Transformation[Hints=>Hints, Service[Alg], Service[Alg]] =
+    (func: Hints => Hints, that: Service[Alg]) => new Service[Alg] {
+      override type Operation[I, E, O, SI, SO] = that.Operation[I, E, O, SI, SO]
+
+      override val endpoints: List[Endpoint[_, _, _, _, _]] = that.endpoints.map(_.mapHints(func))
+
+      override def endpoint[I, E, O, SI, SO](op: Operation[I, E, O, SI, SO]): (I, Endpoint[I, E, O, SI, SO]) = {
+        that.endpoint(op) match {
```

Comment:
The `match` is un-necessary : 

``` 
val (i, endpoint) = that.endpoint(op) 
```

Reply:
isn't that more of a style choice? Or does it differ in behavior?

Reply:
I was using purely because  I like the style for extraction, is there an issue ?

Reply:
I guess it is indeed a stylistic thing, but I have a reasonably objective argument for suggesting that : I think `match` signals either partiality or plurality of patterns. Whenever there is an extraction that is total, I use a `val` extractor, as it makes future-me less prone to eyebrow-raising. Also it's more concise.